File: MERN/week 1/day2/practise/ninja.js
// parent Vehicle class
class Ninja {
    constructor(name) {
        this.name = name;
        this.health = 90;
        this.speed = 3;
        this.strength = 3;
    }
    sayName() {
        console.log (`the ninja is ${ this.name }.`);
    }
    showStats() {
        console.log (`the ninja is ${ this.name }`);
        console.log (`health ${ this.health }`);
        console.log (`speed is ${ this.speed }`);
        console.log (`strength ${ this.strength }`);
    }
    drinkSake() {
        this.health += 10
    }
}



const ninja1 = new Ninja("Hyabusa");
ninja1.sayName();
ninja1.drinkSake();
ninja1.showStats();

File: java/week1/day2/practise/Barista's Challenge/Ninja.java
public class Ninja {
    // Member variables 
    private String name;
    private Integer health; 
    private Weapon weapon;
    private Integer power;


    // constructor 

    public Ninja(String name , Weapon weaponName , Integer power ) {
        this.name = name;
        this.health = 100;
        this.weapon = weaponName;
        this.power = power;
    }

}